src/lib/y2users/linux/users_writer.rb

```diff
@@ -0,0 +1,366 @@
+      end
+
+      # Executes the command for setting the dates and limits in /etc/shadow
+      #
+      # @param user [User]
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def set_password_attributes(user, issues)
+        return unless user.password
+
+        options = chage_options(user)
+
+        return if options.empty?
+
+        Yast::Execute.on_target!(CHAGE, *options, user.name)
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          # TRANSLATORS: %s is a placeholder for a username
+          format(_("Error setting the properties of the password for '%s'"), user.name)
+        )
+        log.error("Error setting password attributes for '#{user.name}' - #{e.message}")
+      end
+
+      # Attributes to modify using `usermod`
+      USERMOD_ATTRS = [:gid, :home, :shell, :gecos].freeze
+
+      # Edits the user
+      #
+      # @param new_user [User] User containing the updated information
+      # @param old_user [User] Original user
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def edit_user(new_user, old_user, issues)
+        usermod_changes = USERMOD_ATTRS.any? do |attr|
+          !new_user.public_send(attr).nil? &&
+            (new_user.public_send(attr) != old_user.public_send(attr))
+        end
+        usermod_changes ||= different_groups?(new_user, old_user)
+
+        Yast::Execute.on_target!(USERMOD, *usermod_options(new_user, old_user)) if usermod_changes
+        change_password(new_user, issues) if old_user.password != new_user.password
+        write_auth_keys(new_user, issues) if old_user.authorized_keys != new_user.authorized_keys
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          format(_("The user '%{username}' could not be modified"), username: new_user.name)
+        )
+        log.error("Error modifying user '#{new_user.name}' - #{e.message}")
+      end
+
+      # Generates and returns the options expected by `useradd` for given user
+      #
+      # @param user [User]
+      # @return [Array<String>]
+      def useradd_options(user)
+        opts = {
+          "--uid"      => user.uid,
+          "--gid"      => user.gid,
+          "--shell"    => user.shell,
+          "--home-dir" => user.home,
+          "--comment"  => user.gecos.join(","),
+          "--groups"   => useradd_groups(user).join(",")
+        }
+        opts = opts.reject { |_, v| v.to_s.empty? }.flatten
+
+        if user.system?
+          opts << "--system"
+        else
+          opts.concat(create_home_options(user))
+        end
+
+        opts << user.name
+        opts
+      end
+
+      # Name of the secondary groups for useradd
+      #
+      # @return [Array<String>]
+      def useradd_groups(user)
+        names = user.secondary_groups_name
+        return names unless names.empty?
```

Comment:
Here I see a bit issue. Can we distinguish between explicit no groups and intentionally skipping of definition of secondary groups?

Reply:
I am not sure if I get your comment that `There is no way in AutoYaST to specify the list of secondary groups for a user (not even to force an empty one).` In https://doc.opensuse.org/projects/autoyast/#Configuration-Security-groups there is attribute userlist that specify secondary groups for given user. Or do I miss something? But I agree that it is hard to distinguish in autoyat itself if there is intentionally no secondary groups or if default should be used :worried:

Reply:
Fun fact. I did some manual testing with the current AutoYaST (before any y2users change) and... ta da! the default list of groups is always ignored.

- If the user is listed in the attribute "userlist" of some group(s), then the internal attribute "grouplist" for that user is initialized according. As a result, the code does not enter [here](https://github.com/yast/yast-users/blob/master/src/modules/Users.pm#L3119). That means the default group list is not used.
- If the user is not listed in any group, the internal attribute "grouplist" is initialized to an empty set. Which means it's defined and the code does not enter the if either. So again the default group list is not used.

We can conclude the "groups" attribute of the `<user_defaults>` section of AutoYaST has been useless since useradd dropped the GROUPS key from `/etc/defaults/useradd`. So the only use case for which we have been implementing all this has disappeared. :disappointed:

Reply:
> We can conclude the "groups" attribute of the `<user_defaults>` section of AutoYaST has been useless since useradd dropped the GROUPS key from `/etc/defaults/useradd`.

In case you wonder. That means with the switch from SLE11 to SLE12 (as documented somewhere in this PR).